OpenProblemLibrary/ma117DB/set5/srw2_9_63.pg

```diff
@@ -20,35 +20,39 @@ DOCUMENT();        # This should be the first executable line in the problem.
 
 loadMacros(
   "PGstandard.pl",
-  "PGchoicemacros.pl",
+  "PGML.pl",
+  "MathObjects.pl",
   "PGcourse.pl"
 );
 
-TEXT(beginproblem());
 $showPartialCorrectAnswers = 1;
 
+Context("Numeric");
+
 $a = random(2,9,1);
+$ans3 = Formula("sqrt(x)-$a");
+
+Context("Interval");
+
+$ansInt = Compute("[-$a, infinity)");
+
+BEGIN_PGML
+The function [`f(x)=(x+[$a])^2`] is not one-to-one. Choose the largest possible
+domain containing the number 100 so that the function restricted to the domain is one-to-one.
+
+The largest possible domain is [_________________]{$ansInt} (use interval notation).
+
+If [`f`] is restricted to the above domain, then the inverse function is [`g(x)=`][_______________]{$ans3}.
+END_PGML
+
+BEGIN_PGML_SOLUTION
+The function [`f(x)=(x+[$a])^2`] is one-to-one on the intervals [`(-\infty, -[$a]]`] and [`[-[$a], \infty)`].
+The second one contains the number 100.
 
-TEXT(EV2(<<EOT));
-The function \(f(x)=(x+$a)^2\) is not one-to-one. Choose a largest possible
-domain containing the number 100 so that 
-the function restricted to the domain is one-to-one.
-$BR  
-The largest possible domain is [\{ ans_rule(10) \},\{ ans_rule(10) \});
-$BR  
-the inverse function is \(g(x)=\) \{ ans_rule(15) \}
-$BR
-$BBOLD
-If your answer is \(\infty\), enter infinity.
-$EBOLD
-EOT
-
-$ans1 = -$a;
-$ans2 = "infinity";
-$ans3 = "sqrt(x)-$a";
-ANS(num_cmp($ans1));
-ANS(str_cmp($ans2));
-ANS(fun_cmp($ans3));
+Solving the equation [`y=(x+[$a])^2`] for [`x`] will give us two solutions: [`x = \pm\sqrt{y} - [$a]`].
+According to our domain choice, [`x \ge [$ans1]`], which means we have to choose the solution with [`+`].
```

Comment:
Also line 53: I'd suggest this wording:  
  The correct sign for the square root is + because [`x \geq [$a]`] in the domain [`[-[$a],\infty)`].